ci-operator/step-registry/assisted/nutanix/setup/assisted-nutanix-setup-commands.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ nutanix assisted test-infra setup machine command ************"
+
+# ensure LEASED_RESOURCE is set
+if [[ -z "${LEASED_RESOURCE}" ]]; then
+  echo "Failed to acquire lease"
+  exit 1
+fi
+
+# shellcheck source=/dev/random
+source $SHARED_DIR/nutanix_context.sh
+
+echo "$(date -u --rfc-3339=seconds) - Getting PE Name"
+
+pc_url="https://${PE_HOST}:${PE_PORT}"
+clusters_api_ep="${pc_url}/api/nutanix/v3/clusters/list"
+un="${NUTANIX_USERNAME}"
+pw="${NUTANIX_PASSWORD}"
+data="{
+  \"kind\": \"cluster\"
+}"
+
+clusters_json=$(curl -ks -u "${un}":"${pw}" -X POST ${clusters_api_ep} -H "Content-Type: application/json" -d @-<<<"${data}")
+pe_name=$(echo "${clusters_json}" | jq '.entities[] | select (.spec.name != "Unnamed") | .spec.name' | head -n 1)
+
+if [[ -z "${pe_name}" ]]; then
+    echo "$(date -u --rfc-3339=seconds) - Cannot get PE Name"
+    exit 1
+fi
+
+subnets_api_ep="${pc_url}/api/nutanix/v3/subnets/list"
+data="{
+  \"kind\": \"subnet\"
+}"
+subnets_json=$(curl -ks -u "${un}":"${pw}" -X POST ${subnets_api_ep} -H "Content-Type: application/json" -d @-<<<"${data}")
+subnet_name=$(echo "${subnets_json}" | jq ".entities[] | .spec.name"  | head -n 1)
+
+echo "$(date -u --rfc-3339=seconds) - PE Name: ${pe_name}"
+echo "$(date -u --rfc-3339=seconds) - Subnet Name: ${subnet_name}"
+
+mkdir -p /home/assisted-test-infra/build/terraform
+cp -r terraform_files/nutanix/* /home/assisted-test-infra/build/terraform/
+cd /home/assisted-test-infra/build/terraform
+
+# Create variables file
+cat >> nutanix-params.hcl << EOF
+nutanix_endpoint = "${NUTANIX_HOST}"
+nutanix_port = "${NUTANIX_PORT}"
+nutanix_username = "${NUTANIX_USERNAME}"
+nutanix_password = "${NUTANIX_PASSWORD}"
+nutanix_cluster = ${pe_name}
+nutanix_subnet = ${subnet_name}
+cluster_name = "assisted-ci-${BUILD_ID}"
+masters_count = 3
+master_memory = 16000
+master_disk = 120000000
+master_vcpu = 4
+workers_count = 2
+worker_memory = 16000
+worker_disk = 120000000
+worker_vcpu = 2
```

Comment:
You don't need those variables

Reply:
Not sure what you mean. These are required to be set.
Should I run some -infra-script to fill them in?